app/javascript/controllers/file_upload_controller.js

```diff
@@ -6,8 +6,17 @@ export default class extends Controller {
 
   select(event) {
     const file = event.currentTarget.files[0]
-    this.imageTarget.src = window.URL.createObjectURL(file)
-    this.imageTarget.srcset = window.URL.createObjectURL(file)
+    const objectUrl = window.URL.createObjectURL(file)
+    this.imageTarget.src = objectUrl
+    this.imageTarget.srcset = objectUrl
+    // NOTE: Update the srcset of the <source> tag of the <picture> tag sub-element
+    const picture = event.currentTarget.closest('div').querySelector('picture');
+    if(picture) {
+      const sources = picture.querySelectorAll('source')
+      sources.forEach(source => {
+        source.srcset = objectUrl
+      });
+    }
```

Comment:
Can we convert this to another Stimulus target? Iterating over `this.pictureTargets` feels a lot cleaner.